rancher/service.go

```diff
@@ -145,7 +146,22 @@ func (r *RancherService) up(create bool) error {
 		service, err = r.context.Client.Service.ActionActivate(service)
 		err = r.Wait(service)
 	}
-
+	for k, v := range r.serviceConfig.Labels {
```

Comment:
Also can you add `// TODO: revisit whether this is the best place to perform this check` above this block?